packages/ai-chat-ui/src/browser/chat-input-widget.tsx

```diff
@@ -215,15 +222,40 @@ const ChatInput: React.FunctionComponent<ChatInputProperties> = (props: ChatInpu
         };
     }, []);
 
+    // track added requests
     React.useEffect(() => {
-        const listener = lastRequest?.response.onDidChange(() => {
-            if (lastRequest.response.isCanceled || lastRequest.response.isComplete || lastRequest.response.isError) {
-                setInProgress(false);
+        const onChatModelUpdate = (event: ChatChangeEvent) => {
+            if (event.kind === 'addRequest') {
+                onAddedRequest(() => event.request);
             }
-        });
+        };
+        const listener = props.chatModel.onDidChange(onChatModelUpdate);
+        return () => listener?.dispose();
+    }, []);
+
+    // remember last requests
+    React.useEffect(() => {
+        if (lastRequest) {
+            setInProgress(ChatRequestModel.isInProgress(lastRequest));
+        }
+        const onResponseUpdate = () => {
+            setInProgress(ChatRequestModel.isInProgress(lastRequest));
+        };
+        const listener = lastRequest?.response.onDidChange(onResponseUpdate);
         return () => listener?.dispose();
     }, [lastRequest]);
 
+    // track change set updates
+    React.useEffect(() => {
+        const onChatModelUpdate = (event: ChatChangeEvent) => {
+            if (ChatChangeEvent.isChangeSetEvent(event)) {
+                forceUpdate();
```

Comment:
It seems the `forceUpdate` is used to update the UI for changes in the changeSet. A bit more elegant would be to maintain a "view model" abstraction in `useState` for the `ChangeSetBox` and update that one whenever needed. Then we would not need a `forceUpdate`